snet_cli/mpe_service_command.py

```diff
@@ -109,18 +115,17 @@ def publish_service_with_metadata(self):
 
     def publish_metadata_in_ipfs_and_update_registration(self):
         # first we check that we do not change payment_address or group_id in existed payment groups
-        if (not self.args.force):
-            old_metadata = self._get_service_metadata_from_registry()
-            new_metadata = load_mpe_service_metadata(self.args.metadata_file)
-            for old_group in old_metadata["groups"]:
-                if (new_metadata.is_group_name_exists(old_group["group_name"])):
-
-                    new_group = new_metadata.get_group(old_group["group_name"])
-                    if (new_group["group_id"] != old_group["group_id"] or new_group["payment_address"] != old_group["payment_address"]):
-                        raise Exception("You are trying to change group_id or payment_address in group '%s'.\n"%old_group["group_name"] +
-                                         "You will make all open channels invalid.\n" +
-                                         "Use --force if you really want it.")
-
+        old_metadata = self._get_service_metadata_from_registry()
+        new_metadata = load_mpe_service_metadata(self.args.metadata_file)
+        for old_group in old_metadata["groups"]:
+            if (new_metadata.is_group_name_exists(old_group["group_name"])):
```

Comment:
Can it happen that the group names change completely across a publish. i.e. I have g1 and g2 in old_metadata and g3 and g4 in new_metadata

Reply:
Yes, it is possible. I only check that ```group_id``` or ```payment_address``` haven't been changed in the same groups. It will work in most cases because almost everybody have only one `default_group`.

We also probably shoudn't forbid to remove payment groups...

Reply:
So shouldnt we do the same check where group names have been changed completely? In that scenario too open channels would become invalid.

Reply:
In principle we could forbid to removing payment_group (via snet-cli). There is always a possibility to simply remove service and register it again....
So should I forbid to removing group?

Reply:
Preventing developers from removing payment groups seems restrictive for me. Imagine that I had set up two groups corresponding to two different daemon installations and now decided to run just one group, I should have the flexibility to remove or disable the other. Lets go ahead and merge this request in. We can track an issue to handle the scenario where groups have changed.

Reply:
ok! so I merge it.